src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -79,6 +80,24 @@ trait Producer[T] extends PipelineStep with CachingEnabled with Logging {
   ): PersistedProducer[T, A] =
     new PersistedProducer(this, io, artifactSource)
 
+  /** Wrap the Producer into a PersistedProducer, which will persist the data when using .get
+    * method.
+    */
+  // REVIEW data is essentially io + artifactSource, rolled into one. However, flexibility
+  // makes it increasingly hard for users to figure out how to configure the system.
+  // I'm not aware of a use-case that requires changing both the serialization and the "artifact"
+  // independently. Even if there is, chances are it can be hidden behind a properly constructed
+  // PartialFileItem.
+  def persisted(metaFs: FlatFileSystem, data: PartialFileItem[T]) = {
```

Comment:
You're passing in a `FlatFileSystem`, which has a method that returns a `PartialFileItem` but also passing in a `PartialFileItem` instance?  Doesn't make sense.  Fundamentally, you're passing in a factory so that the `PersistedProducer` can create the `FileItem` from it, but I think you should just pass in the `FileItem` itself.  Whether the `FileItem` was created directly or via a factory doesn't matter to the `Producer`